### PR TITLE
Correction within the About apps page

### LINKED ISSUE #4553: Page links to itself
### What article on docs.github.com is affected?

[About Apps](https://docs.github.com/en/developers/apps/about-apps)

### What part(s) of the article would you like to see updated?

In the second paragraph of the article about apps, there is a link which links to itself.

>... For information on choosing a type of app, see "[About Apps](https://docs.github.com/en/developers/apps/about-apps)
" and ... "

### Additional information

None

### PR DESCRIPTION
**Pull request by lucalves**
### Why:

I removed a link from an article that was directing itself and updated the article "Differences between apps" to "Differences between GitHub Apps and OAuth Apps.", suggestion sent by @janiceilene at #4553.

**Closes #4553**.

### Check off the following:
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: content/developers/apps/about-apps.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ versions:
 
 Apps on {% data variables.product.prodname_dotcom %} allow you to automate and improve your workflow. You can build apps to improve your workflow.{% if currentVersion == "free-pro-team@latest" %} You can also share or sell apps in [{% data variables.product.prodname_marketplace %}](https://github.com/marketplace). To learn how to list an app on {% data variables.product.prodname_marketplace %}, see "[Getting started with GitHub Marketplace](/marketplace/getting-started/)."{% endif %}
 
-{% data reusables.marketplace.github_apps_preferred %}, but GitHub supports both {% data variables.product.prodname_oauth_app %}s and {% data variables.product.prodname_github_apps %}. For information on choosing a type of app, see "[About apps](/apps/about-apps/)" and "[Differences between apps](/apps/differences-between-apps/)."
+{% data reusables.marketplace.github_apps_preferred %}, but GitHub supports both {% data variables.product.prodname_oauth_app %}s and {% data variables.product.prodname_github_apps %}. For information on choosing a type of app, see "[Differences between GitHub Apps and OAuth Apps](/apps/differences-between-github-apps-and-oauth-apps/)."
```
Comment (janiceilene) [inline content/developers/apps/about-apps.md:17]:
```suggestion
{% data reusables.marketplace.github_apps_preferred %}, but GitHub supports both {% data variables.product.prodname_oauth_app %}s and {% data variables.product.prodname_github_apps %}. For information on choosing a type of app, see "[Differences between GitHub Apps and OAuth Apps](/developers/github-marketplace/about-github-marketplace)."
```

Thanks so much for opening a PR with this fix @lucalves! When I go to the staged article https://docs-4590--hotfixabout-app-lin.herokuapp.com/en/developers/apps/about-apps and click on the new, it gives an error. 

I think we'll need to update the relative link to `/developers/github-marketplace/about-github-marketplace`. If you look up at line 7 in this diff, you can see that the old link was redirecting to the current article, so I think this updated link should work!

Let me know when you accept this change and we can double check that the link works and get this merged in 🎉

**Comment by lucalves** [comment]
Hello, @janiceilene! I committed the suggestions. I believe it is now working as expected.

File: content/developers/apps/about-apps.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ versions:
 
 Apps on {% data variables.product.prodname_dotcom %} allow you to automate and improve your workflow. You can build apps to improve your workflow.{% if currentVersion == "free-pro-team@latest" %} You can also share or sell apps in [{% data variables.product.prodname_marketplace %}](https://github.com/marketplace). To learn how to list an app on {% data variables.product.prodname_marketplace %}, see "[Getting started with GitHub Marketplace](/marketplace/getting-started/)."{% endif %}
 
-{% data reusables.marketplace.github_apps_preferred %}, but GitHub supports both {% data variables.product.prodname_oauth_app %}s and {% data variables.product.prodname_github_apps %}. For information on choosing a type of app, see "[About apps](/apps/about-apps/)" and "[Differences between apps](/apps/differences-between-apps/)."
+{% data reusables.marketplace.github_apps_preferred %}, but GitHub supports both {% data variables.product.prodname_oauth_app %}s and {% data variables.product.prodname_github_apps %}. For information on choosing a type of app, see "[Differences between GitHub Apps and OAuth Apps](/developers/github-marketplace/about-github-marketplace)."
```
Comment (janiceilene) [inline content/developers/apps/about-apps.md:17]:
```suggestion
{% data reusables.marketplace.github_apps_preferred %}, but GitHub supports both {% data variables.product.prodname_oauth_app %}s and {% data variables.product.prodname_github_apps %}. For information on choosing a type of app, see "[Differences between GitHub Apps and OAuth Apps](/developers/apps/differences-between-github-apps-and-oauth-apps)."
```

**Review by janiceilene — APPROVED** [review]
Thanks so much @lucalves! I'll update the branch and get this merged in once tests are passing 🍏

**Pull request merged**